Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m

```diff
@@ -78,26 +75,19 @@ - (BOOL)hasQuotedAttachment
         && ![OWSMimeTypeOversizeTextMessage isEqualToString:self.quotedMessage.contentType]);
 }
 
-- (BOOL)hasQuotedAttachmentThumbnail
+- (BOOL)hasQuotedAttachmentThumbnailImage
 {
     return (self.quotedMessage.contentType.length > 0
         && ![OWSMimeTypeOversizeTextMessage isEqualToString:self.quotedMessage.contentType] &&
         [TSAttachmentStream hasThumbnailForMimeType:self.quotedMessage.contentType]);
 }
 
-- (NSString *)quotedSnippet
+- (UIColor *)highlightColor
 {
-    if (self.displayableQuotedText.displayText.length > 0) {
-        return self.displayableQuotedText.displayText;
-    } else {
-        NSString *mimeType = self.quotedMessage.contentType;
-
-        if (mimeType.length > 0) {
-            return [TSAttachment emojiForMimeType:mimeType];
-        }
-    }
-
-    return @"";
+    BOOL isIncomingQuote
```

Comment:
Many of the changes in this PR bring us closer to the spec.  I'm hoping to review with Myles in detail today.